src/scripting/lua_unit.cpp

```diff
@@ -239,6 +239,28 @@ static int impl_unit_equality(lua_State* L)
 	return 1;
 }
 
+/**
+ * Turns a lua proxy unit to string. (__tostring metamethod)
+ */
+static int impl_unit_tostring(lua_State* L)
+{
+	const unit& u = luaW_checkunit(L, 1);
+	std::ostringstream str;
+
+	str << "unit: <";
+	if(!u.id().empty()) {
+		str << u.id() << " ";
+	} else {
+		str << u.type_id() << " ";
+	}
+	// This will say 0,0 (or -999,-999?) for units on the recall list
+	str << "at (" << u.get_location() << ")";
```

Comment:
@CelticMinstrel suggests to say "on side N recall list" when the unit is on a recall list.

Reply:
are ou sure it will always return 999,999 for recall list units? I currently don't see the code that resets the location when moving a unit to the recall list.

Reply:
-999,-999 is what goto_x,goto_y are set to in some cases. I currently don't see how to detect a unit being on a recall list, other than iterate all teams and check for each team's recall list whether it contains the unit by id.

A unit that starts out on the recall list with `{UNIT 1 Mage recall recall ()}` has location=(0,0) in lua.

Reply:
Sorry, when I said "in lua" I meant "when printed from the C++ implementation of the Lua API", that's rather different.